### DISCUSSION TITLE
Working Great for Colmap to Blender!

### DISCUSSION THREAD

**Discussion by quitmeyer** [SBCV/Blender-Addon-Photogrammetry-Importer#65] [Show and tell]
I am trying to build a kind of 3D scanning and then projection mapping system with colmap and blender, and just wanted to say your plugin has been working really great!

We have a virtual camera that simulates the projector, and then try to align it in blender to change the scene and add stuff to the real world place. We almost have the system running, but still working on getting stuff lined up! but your plugin has been super helpful!
![image](https://github.com/SBCV/Blender-Addon-Photogrammetry-Importer/assets/742627/5aa2ee71-cb09-469b-9431-bdf7a947586b)

right now we are having a tricky part trying to get a totally perfect alignment (because it seems we have to use a simpler model like SIMPLE PINHOLE) or we have to add distortion via the movie distort thing in blender's compositor (which none of the options there seem to perfectly line up with what blender offers for distortion models).

quick question for you though might be, is it possible to use this plugin with camera models that have distinct Fx Fy focal lengths? like PINHOLE instead of SIMPLE_PINHOLE?

**Comment by SBCV** [comment]
The last time I checked Blender did not support camera models with different focal length values in x and y direction - this is why the addon does not support other camera models. I do not know if this has been changed in the meantime, or if there is a workaround available.

However, what you can always convert a pinhole camera (e.g. containing fx and fy) to another pinhole camera (e.g. containing only fx) by transforming the corresponding image data. See for example section 4.1 and 4.2 in [this paper](https://isprs-archives.copernicus.org/articles/XLIII-B2-2021/313/2021/isprs-archives-XLIII-B2-2021-313-2021.pdf).
